### PR TITLE
Using the new HttpClient API in Credentials

### PR DESCRIPTION
**Pull request by hiranya911**
Using the new `HttpClient` API in Credentials module. With this I believe we are using the new API everywhere in the SDK (I will create a separate PR to remove the old code). I also managed to re-enable an old refresh token unit test that had been disabled for a long time.

This is a follow up to #280, #291, and #301

### CODE REVIEW

File: src/auth/credential.ts [hunk 1]
```diff
@@ -186,75 +185,59 @@ export interface GoogleOAuthAccessToken {
...
+    );
   });
 }
 
 /**
  * Implementation of Credential that uses a service account certificate.
  */
 export class CertCredential implements Credential {
-  private certificate_: Certificate;
+  private readonly certificate_: Certificate;
```
Comment (bojeil-google) [inline src/auth/credential.ts]: since you're already modifying this, let's remove the trailing underscore.

File: src/auth/credential.ts [hunk 2]
```diff
@@ -295,11 +278,13 @@ export interface Credential {
  * Implementation of Credential that gets access tokens from refresh tokens.
  */
 export class RefreshTokenCredential implements Credential {
-  private refreshToken_: RefreshToken;
+  private readonly refreshToken_: RefreshToken;
```
Comment (bojeil-google) [inline src/auth/credential.ts]: Same here. Let's remove trailing underscore.

File: src/auth/credential.ts (hunk 1 above)
Comment (hiranya911) [reply src/auth/credential.ts]: Done

File: src/auth/credential.ts (hunk 2 above)
Comment (hiranya911) [reply src/auth/credential.ts]: Done

**Pull request merged**